### ISSUE TITLE
VO on Embed links in emails don't read the URL which can cause trust issues

### ISSUE THREAD

**Issue by willeybryan** [cds-snc/notification-api#395]
**What?**
- Password reset email has a link to 'password reset' which hides the URL behind it
- VO screen readers should be able to read this URL some how even if not displayed, not just for password email, but in general

**Why?**
- If a user gets an email they weren't expecting they will want to scrutinize it to ensure it isn't spam

**Comment by timarney** [comment]
Can we document somewhere how we want to treat email links?  I'm unclear at this point.

Given the following link as an example (for the way which we should treat all email links)

```
https://notification.alpha.canada.ca/forgot-password
```

Keeping in mind emails address with tokens
```
https://notification.alpha.canada.ca/email-auth/%some-really-long-random-token-here
```

Email should show?

a) Forgot password
b) https://notification.alpha.canada.ca/forgot-password
c)  notification.alpha.canada.ca/forgot-password

cc: @JuliannaR @maxneuvians 

- Differences between screen readers etc... 

Do we need testing to validate?

**Comment by timarney** [comment]
After doing some additional research of this I don't think there much we can do about this behaviour.

Testing with VoiceOver on Mac OS (that's all I have to test with)

Using the VoiceOver rotor menu + regular screen reader navigation link text is read but not the link behind the text.  This seems to be true of web pages as well (not just emails).

Trying to get a screen reader to read the original email source is a really bad experience (at least in gmail).

I found some link scanning sites as well as lots of recommendations for malware software etc... that claim to detect phishing links.

Trying to find content for a11y and phishing links is challenging as well.

**Comment by timarney** [comment]
Also just for tracking purposes Gmail flags emails as well.

<img width="916" alt="Screen Shot 2019-10-16 at 7 54 10 AM" src="https://user-images.githubusercontent.com/62242/66919767-fb121800-efef-11e9-862a-4ef20508d819.png">

**Comment by willeybryan** [comment]
Thanks for looking into this. @JuliannaR I am not sure there is more we can do.

I am going to ensure our template writing guide contains best practices for diff use cases
a) Forgot password and other expected CTA generated by the user (Expected, this can be a hidden link b/c reading token is bad UX)
b) Unexpected Links to CTA (Expand them so people can see them)
c) Links to more information (Expand them so people can review where they go)

**Comment by timarney** [comment]
Also for tracking:

1% traction on the Twitter research.

https://twitter.com/timarney/status/1184134135352311809

**Comment by JuliannaR** [comment]
There is little documentation on can find on a11y and phishing in emails and I think that is probably a big disconnect on experience.
For JAWS the user is able to list all links available on a page, the problem with gmail is that there is an abundance of links. I think it's important to be able to identify the path that someone is going too. But I agree this is difficult to assess. When we do usability testing, I'd like to ensure we review this specifically in a research activity.

**Comment by timarney** [comment]
@willeybryan do you want to open a different research ticket for this?

**Comment by willeybryan** [comment]
@timarney Yep, research and design is aware of this and we are working to find more info about emails. 

Closing for now. Thanks everyone.
